app/scripts/modules/core/src/pipeline/config/stages/manualJudgment/ManualJudgmentApproval.tsx

```diff
@@ -29,17 +33,60 @@ export class ManualJudgmentApproval extends React.Component<
       submitting: false,
       judgmentDecision: null,
       judgmentInput: {},
+      applicationRoles: {},
+      userRoles: [],
       error: false,
     };
   }
 
+  public componentDidMount() {
+    const applicationName = this.props.execution.application;
+    ApplicationReader.getApplicationPermissions(applicationName).then(result => {
+      if (typeof result !== 'undefined') {
+        this.setState({
+          applicationRoles: result,
+        });
+        this.isManualJudgmentStageNotAuthorized();
+      }
+    });
+    this.setState({
+      userRoles: AuthenticationService.getAuthenticatedUser().roles,
+    });
+  }
+
   private provideJudgment(judgmentDecision: string): void {
     const { application, execution, stage } = this.props;
     const judgmentInput: string = this.state.judgmentInput ? this.state.judgmentInput.value : null;
     this.setState({ submitting: true, error: false, judgmentDecision });
     ReactInjector.manualJudgmentService.provideJudgment(application, execution, stage, judgmentDecision, judgmentInput);
   }
 
+  private isManualJudgmentStageNotAuthorized(): boolean {
+    let isStageNotAuthorized = true;
+    let usrRole;
+    const stageRoles = this.props.stage.context.selectedStageRoles || [];
+    const readArray = this.state.applicationRoles['READ'] || [];
+    const writeArray = this.state.applicationRoles['WRITE'] || [];
+    const executeArray = this.state.applicationRoles['EXECUTE'] || [];
+    const createArray = this.state.applicationRoles['CREATE'] || [];
+    const usrRoles = this.state.userRoles;
+    if (stageRoles.length === 0) {
+      isStageNotAuthorized = false;
+      return isStageNotAuthorized;
+    }
+    for (usrRole of usrRoles) {
+      if (stageRoles.includes(usrRole)) {
+        if (writeArray.includes(usrRole) || executeArray.includes(usrRole) || createArray.includes(usrRole)) {
+          isStageNotAuthorized = false;
+          return isStageNotAuthorized;
+        } else if (readArray.includes(usrRole)) {
+          isStageNotAuthorized = true;
+        }
+      }
+    }
+    return isStageNotAuthorized;
+  }
+
```

Comment:
I think we can optimize the function a bit. It looks like all you care about is if the user can write, execute or create. Please correct me if I am wrong but the logic below considers that case. 

Note, the logic flips since we are checking if the user **is** authorized, since that would be the happy path, and special handling is needed when they are not authorized. 

```js
private isUserAuthorizedForManualJudgment(): boolean {
     const { applicationRoles, userRoles } = this.state;
     const stageRoles = this.props.stage.context.selectedStageRoles || [];

     if (!stageRoles.length) {
        return true;
     }
    
    let isAuthorized = true; 
    userRoles.forEach( userRole => {
         if (stageRoles.includes(userRole)) {
           const { CREATE, EXECUTE, READ, WRITE } = applicationRoles;
           isAuthorized = (WRITE || []).includes(userRole) || (EXECUTE || []).includes(userRole) || (CREATE || []).includes(userRole);
          }
      })

     return isAuthorized;
    }    
}
```